OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs

```diff
@@ -105,23 +101,16 @@ public void SendAirstrike(Actor self, WPos target, bool randomize = true, int at
 
 				// Remove the camera when the final plane leaves the target area
 				if (!aircraftInRange.Any(kv => kv.Value))
-				{
-					if (camera != null)
-					{
-						camera.QueueActivity(new Wait(info.CameraRemoveDelay));
-						camera.QueueActivity(new RemoveSelf());
-					}
-
-					camera = null;
+					RemoveCamera(camera);
+			};
 
-					if (beacon != null)
-					{
-						self.World.AddFrameEndTask(w =>
-						{
-							w.Remove(beacon);
-							beacon = null;
-						});
-					}
+			Action<Actor> onRemovedFromWorld = a =>
+			{
+				// Remove the camera and beacon if whole squad is shot down
```

Comment:
I don't understand why
```csharp
aircraftInRange[a] = false;
if (!aircraftInRange.Any(kv => kv.Value))
```
doesn't work here, which would simplify the fix to moving the `RemoveBeacon` call inside the `if` inside `onExitRange`.  Can you please expand this comment explaining why the check needs to be different, or re-merge the actions if it doesn't?

Reply:
Handling `onExitRange` and `onRemovedFromWolrd` in the same place leads to the bug that I set out to fix here. If an aircraft is shot down before the squad has reached target range that check will return `true` because all of the aircraft will be out of range. That's why checking for attack range is not relevant when aircraft actors are removed from world. That's why I'm checking if all aircraft `isDead`. 

Another way to go about it may be to remove actors from the dictionary when they killed and remove the camera and beacon once the dictionary is empty.

Reply:
Aha, now I see.

> Another way to go about it may be to remove actors from the dictionary when they killed and remove the camera and beacon once the dictionary is empty.

This sounds good to me

Reply:
I have expanded the comment.